lerobot/common/datasets/lerobot_dataset.py

```diff
@@ -147,6 +150,158 @@ def __repr__(self):
+            )
+
+        if isinstance(test_size, float):
+            n_test = ceil(test_size * self.num_episodes)
+        elif isinstance(test_size, int):
+            n_test = float(test_size)
+
+        if isinstance(train_size, float):
+            n_train = floor(train_size * self.num_episodes)
+        elif isinstance(train_size, int):
+            n_train = float(train_size)
+
+        if train_size is None:
+            n_train = self.num_episodes - n_test
+        elif test_size is None:
+            n_test = self.num_episodes - n_train
+
+        if n_train + n_test > self.num_episodes:
+            raise ValueError(
+                f"The sum of train_size and test_size = {n_train + n_test}, "
+                "should be smaller than the number of "
+                f"samples {self.num_episodes}. Reduce test_size and/or "
+                "train_size."
+            )
+
+        n_train, n_test = int(n_train), int(n_test)
+
+        if n_train == 0:
+            raise ValueError(
+                f"With self.num_episodes={self.num_episodes}, test_size={test_size} and train_size={train_size}, the "
+                "resulting train set will be empty. Adjust any of the "
+                "aforementioned parameters."
+            )
+        
+        if not shuffle:
+            train_episode_indices = np.arange(n_train)
+            test_episode_indices = np.arange(n_train, n_train + n_test)
+        else:
+            permutation = np.random.permutation(self.num_episodes)
+            test_episode_indices = permutation[:n_test]
+            train_episode_indices = permutation[n_test : (n_test + n_train)]
+
+        train_indices = [idx for idx, episode_idx in enumerate(self.hf_dataset["episode_index"]) if episode_idx.item() in train_episode_indices]
+        test_indices = [idx for idx, episode_idx in enumerate(self.hf_dataset["episode_index"]) if episode_idx.item() in test_episode_indices]
+
+        train_split = LeRobotDataset.from_preloaded(
+            repo_id=self.repo_id,
+            version=self.version,
+            root=self.root,
+            split=self.split,
+            transform=self.transform,
+            delta_timestamps=self.delta_timestamps,
+            hf_dataset=self.hf_dataset.select(indices=train_indices),
+            stats=self.stats,
+            info=self.info,
+            videos_dir=self.videos_dir,
+        )
+        train_split.create_episode_data_index()
+
+        test_split = LeRobotDataset.from_preloaded(
+            repo_id=self.repo_id,
+            version=self.version,
+            root=self.root,
+            split=self.split,
+            transform=self.transform,
+            delta_timestamps=self.delta_timestamps,
+            hf_dataset=self.hf_dataset.select(indices=test_indices),
+            stats=self.stats,
+            info=self.info,
+            videos_dir=self.videos_dir,
+        )
+        test_split.create_episode_data_index()
+
+        return train_split, test_split
+
```

Comment:
I think it's possible to find the frame id of a certain episode in `episode_data_index" and use:
https://huggingface.co/docs/datasets/v2.19.0/loading#slice-splits

Reply:
Hey @Cadene! I am not entirely sure I follow your plan with regards to looking up the frame id. More pointers would be appreciated.

But since I wrote the long comment I thought it might be easier if I just implemented in code what I had in mind.

So we have this now :slightly_smiling_face: 
![image](https://github.com/huggingface/lerobot/assets/2444926/191cd9b9-c4a2-41a2-bd87-2d23eb09cbc2)

Super simple, requires less code for saving and loading the dataset, and lets the user do whatever they'd like.

Happy to continue iterating on this with your feedback

Reply:
Mhmm seems there are a couple of other places in code that rely on the episode index containing tensors...

Plus need to take a closer look at `load_previous_and_future_frames`

Reply:
fixed things in a bunch of places across the repo, ran tests, fixed what I could

unfortunately, tests on `main` are failing as well, I am not sure what the failures are due to, suspect some might be due to my env but not sure (I recreated a conda env and followed the instructions from the readme doing `pip install ".[aloha, pusht, xarm]"`)